lib/licensed/dependency.rb

```diff
@@ -68,24 +68,24 @@ def local_files
     # Resets all local project and license information
     def reset_license!
       @project = nil
-      @matched_project_file = nil
+      @project_license = nil
       self.delete("license")
       self.text = nil
     end
 
-    # Returns the Licensee::ProjectFile representing the matched_project_file
-    # or remote_license_file
-    def project_file
-      matched_project_file || remote_license_file
-    end
-
-    # Returns the Licensee::LicenseFile, Licensee::PackageManagerFile, or
-    # Licensee::ReadmeFile with a matched license, in that order or nil
-    # if no license file matched a known license
-    def matched_project_file
-      @matched_project_file ||= project.matched_files
-                                       .select { |f| f.license && !f.license.other? }
-                                       .first
+    # Finds and returns a license from the Licensee::FSProject for this dependency.
+    def project_license
+      @project_license ||= if project.license && !project.license.other?
+        # if Licensee has determined a project to have a specific license, use it!
+        project.license
+      elsif project.license_files.map(&:license).uniq.size > 1
+        # if there are multiple license types found from license files,
+        # return 'other'
+        Licensee::License.find('other')
```

Comment:
Could the elsif branch be deleted if `&& !project.license.other?` were also deleted?

Reply:
I don't think it can.  The call chain for  callstack for this would look like
- `Project#license`
- `Project#licenses_without_copyright`
- `Project#matched_files`
- `Project#project_files`

`#project_files` pulls from `#license_files`, `#package_manager_file` and `#readme_file`.  For the purposes of this PR I was calling out a special case when there were multiple, conflicting, license files.

Note that this doesn't include a scenario where there are multiple license files but they're all other.  In that scenario there is still only a single license found from license files and we fall into the `else` case to look at the package manager and readme files.

Reply:
> For the purposes of this PR I was calling out a special case when there were multiple, conflicting, license files.

Doesn't licensee handle that case directly by itself?

https://github.com/benbalter/licensee/blob/d606b06860fd290693280060faee9c61cbf41c4b/lib/licensee/projects/project.rb#L26-L27

Reply:
It's really similar, but not exactly the same.  What you linked is evaluating licenses from license files, package manager files and readme files.  The one in my change is only looking at [license files](https://github.com/benbalter/licensee/blob/d606b06860fd290693280060faee9c61cbf41c4b/lib/licensee/project_files/license_file.rb#L26-L43), where I use the term to mean files matching that regex e.g. LICENSE or COPYING.

Before this change, `licensed` used the first `project.matched_files` file that had a non-other license.  The issue I saw was that when using multiple license files, is it ok to say that one of them is more relevant than another?

For instance if there are both `LICENSE` and `COPYING` files that contain different licenses, is it ok to say that `LICENSE` takes precedence over `COPYING`?

The logic here is meant to check this scenario.  It makes the assumption that rather than checking `[license file 1, license file 2, ... , package manager file, README]` one at a time as `licensed` was previously doing, it should be checking `[[all license files], package manager file, README]`.

Though based on ☝️ I should probably replace the existing `if` check with `project.license_files.map(&:license).uniq.size == 1`.

Does that make any more sense?  💭 ?

Reply:
> Does project.license there not evaluate licenses from license files, package manager files and readme files?

`project.license` from the if check there does, but `project.license_files` from the elsif check does not.

> I'm not sure when the code in 81-84 would obtain a different result, except where project.license returned other due to conflicting info in non-license files rather than multiple license files with conflicting info, but in that case other is still the right answer.

Yep!  That's exactly the case I was trying to describe.  I don't know that my logic is correct either in implementation or intentions but maybe an example would be better.

Looking at the tests, I think the relevant test cases that describe this logic are [finding licenses from non-license files](https://github.com/github/licensed/blob/a521ff925a77dc62e7366ac45461ec22cfd60c26/test/dependency_test.rb#L56-L75) and [how to handle license files that don't match](https://github.com/github/licensed/pull/64/files/4f04102e07388a2926d8362485aacd40641ee0d7#diff-af404304c2cc2c097e31ce95b4104687R142).  Are the outcomes correct for those scenarios?

Reply:
I made the logic a bit more conservative in ac4b4ce. If there are any potentially conflicting license indicators, we get `other`, requiring human review. Not attached to implementation of course.